### PR TITLE
Add teams documentation for providers that support custom tags

### PR DESCRIPTION
**Pull request by Aaron-9900**
### What does this PR do? What is the motivation?

Adds documentation for teams support.

Added only to providers that support custom tags, taken from here: https://datadoghq.atlassian.net/wiki/spaces/CIAPP/pages/2205614232/CI+Provider+integrations+feature+support

### Merge instructions

- [ ] Pending release of https://datadoghq.atlassian.net/browse/CIAPP-6717
- [ ] Please merge after reviewing

### Additional notes

Preview: https://docs-staging.datadoghq.com/aaron.hoffman/CIAPP-7882_add-teams-documentation/continuous_integration/pipelines/#use-ci-pipelines-data

### CODE REVIEW

File: content/en/continuous_integration/pipelines/circleci.md [hunk 1]
```diff
@@ -49,6 +49,9 @@ To set custom tags to all the pipeline and job spans generated by the integratio
 
 `?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`
 
+#### Integrate with Datadog teams
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/circleci.md:52]: Can we capitalize to be Datadog Teams?

File: content/en/continuous_integration/pipelines/circleci.md [hunk 2]
```diff
@@ -49,6 +49,9 @@ To set custom tags to all the pipeline and job spans generated by the integratio
...
+To display and filter the teams associated with your pipelines, add `team:<your-team>` as a custom tag.
```
Comment (kassenq) [inline content/en/continuous_integration/pipelines/circleci.md:53]: Let's add a sentence after: 
The custom tag name must exactly match your [Datadog Teams](https://docs.datadoghq.com/account_management/teams/) team handle.

File: content/en/continuous_integration/pipelines/circleci.md (hunk 1 above)
Comment (kassenq) [reply content/en/continuous_integration/pipelines/circleci.md:52]: Apply this change to all pages

File: content/en/continuous_integration/pipelines/circleci.md (hunk 2 above)
Comment (kassenq) [reply content/en/continuous_integration/pipelines/circleci.md:53]: Apply this change to all pages

File: content/en/continuous_integration/pipelines/circleci.md (hunk 1 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/circleci.md:52]:
```suggestion
#### Integrate with Datadog Teams
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 3]
```diff
@@ -151,6 +151,9 @@ To set custom tags to all the pipeline and job spans generated by the integratio
 
 `?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`
 
+#### Integrate with Datadog teams
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/gitlab.md:154]:
```suggestion
#### Integrate with Datadog Teams
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 4]
```diff
@@ -739,6 +739,9 @@ datadog(tags: ["team:backend", "release:canary"]){
 }
 {{< /code-block >}}
 
+#### Integrate with Datadog teams
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/jenkins.md:742]:
```suggestion
#### Integrate with Datadog Teams
```

File: content/en/continuous_integration/pipelines/circleci.md (hunk 2 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/circleci.md:53]:
```suggestion
To display and filter the teams associated with your pipelines, add `team:<your-team>` as a custom tag. The custom tag name must match your [Datadog Teams][8] team handle exactly.
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 5]
```diff
@@ -151,6 +151,9 @@ To set custom tags to all the pipeline and job spans generated by the integratio
...
+To display and filter the teams associated with your pipelines, add `team:<your-team>` as a custom tag.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/gitlab.md:155]:
```suggestion
To display and filter the teams associated with your pipelines, add `team:<your-team>` as a custom tag. The custom tag name must match your [Datadog Teams][8] team handle exactly.
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 6]
```diff
@@ -739,6 +739,9 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+To display and filter the teams associated with your pipelines, add `team:<your-team>` as a custom tag.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/jenkins.md:743]:
```suggestion
To display and filter the teams associated with your pipelines, add `team:<your-team>` as a custom tag. The custom tag name must match your [Datadog Teams][8] team handle exactly.
```

**Review by alai97 — APPROVED** [review]

**Comment by alai97** [comment]
Hey @Aaron-9900 or @kassenq , feel free to ping me in Slack when you're ready to merge!

**Pull request merged**